platform/commonUI/edit/src/actions/RemoveAction.js

```diff
@@ -23,111 +23,121 @@
 /**
  * Module defining RemoveAction. Created by vwoeltje on 11/17/14.
  */
-define(
-    [],
-    function () {
+define([
+    './RemoveDialog'
+], function (
+    RemoveDialog
+) {
 
-        /**
-         * Construct an action which will remove the provided object manifestation.
-         * The object will be removed from its parent's composition; the parent
-         * is looked up via the "context" capability (so this will be the
-         * immediate ancestor by which this specific object was reached.)
-         *
-         * @param {DomainObject} object the object to be removed
-         * @param {ActionContext} context the context in which this action is performed
-         * @memberof platform/commonUI/edit
-         * @constructor
-         * @implements {Action}
+    /**
+     * Construct an action which will remove the provided object manifestation.
+     * The object will be removed from its parent's composition; the parent
+     * is looked up via the "context" capability (so this will be the
+     * immediate ancestor by which this specific object was reached.)
+     *
+     * @param {DialogService} dialogService a service which will show the dialog
+     * @param {NavigationService} navigationService a service that maintains the current navigation state
+     * @param {ActionContext} context the context in which this action is performed
+     * @memberof platform/commonUI/edit
+     * @constructor
+     * @implements {Action}
+     */
+    function RemoveAction(dialogService, navigationService, context) {
+        this.domainObject = (context || {}).domainObject;
+        this.dialogService = dialogService;
+        this.navigationService = navigationService;
+    }
+
+    /**
+     * Perform this action.
+     *
+     * @returns {RemoveDialog}
```

Comment:
@tobiasbrown JSdoc needs to be updated to reflect no return value.

Reply:
Thanks @larkin.  I've removed the `@returns` tag.